modular_skyrat/master_files/code/modules/client/preferences/antag_opt_in_preferences.dm

```diff
@@ -0,0 +1,115 @@
+/datum/config_entry/flag/disable_antag_opt_in_preferences
+	default = FALSE
+
+/datum/preference/toggle/master_antag_opt_in_preferences
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "master_antag_opt_in_pref"
+	savefile_identifier = PREFERENCE_PLAYER
+	default_value = TRUE
+
+/datum/preference/toggle/master_antag_opt_in_preferences/is_accessible(datum/preferences/preferences)
+	if (!..(preferences))
+		return FALSE
+
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+
+	return TRUE
+
+/datum/preference/toggle/master_antag_opt_in_preferences/deserialize(input, datum/preferences/preferences)
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+	. = ..()
+
+/datum/preference/toggle/antag_opt_in
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_identifier = PREFERENCE_PLAYER
+	savefile_key = "antag_opt_in_pref"
+	default_value = FALSE
+
+/datum/preference/toggle/antag_opt_in/is_accessible(datum/preferences/preferences)
+	if (!..(preferences))
+		return FALSE
+
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+
+	return preferences.read_preference(/datum/preference/toggle/master_antag_opt_in_preferences)
+
+/datum/preference/toggle/antag_opt_in/deserialize(input, datum/preferences/preferences)
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+	if(!preferences.read_preference(/datum/preference/toggle/master_antag_opt_in_preferences))
+		return FALSE
+	. = ..()
+
+/datum/preference/toggle/antag_opt_in/apply_to_client_updated(client/client, value)
+	. = ..()
+	var/mob/living/carbon/human/target = client?.mob
+	if(!value && istype(target))
+		target.arousal = 0
+		target.pain = 0
+		target.pleasure = 0
+	client.mob.hud_used.hidden_inventory_update(client.mob)
+	client.mob.hud_used.persistent_inventory_update(client.mob)
+
+/datum/preference/choiced/antag_opt_in_status
+	category = PREFERENCE_CATEGORY_NON_CONTEXTUAL
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "antag_opt_in_status_pref"
+
+/datum/preference/choiced/antag_opt_in_status/init_possible_values()
```

Comment:
```suggestion
/datum/preference/toggle/antag_opt_in_status/init_possible_values()
    default_value = FALSE
```

no reason to use a choiced (pick from a list) if youre only using yes/no (which is a toggle)

Reply:
oh wait ignore the default_value, that needs to go on the type